app/models/package_manager/pypi.rb

```diff
@@ -11,6 +11,8 @@ class Pypi < Base
     ENTIRE_PACKAGE_CAN_BE_DEPRECATED = true
     SUPPORTS_SINGLE_VERSION_UPDATE = true
     PYPI_PRERELEASE = /(a|b|rc|dev)[0-9]+$/.freeze
+    # Adapted from https://peps.python.org/pep-0508/#names
+    PYPI_508_NAME_REGEX = /^([A-Z0-9][A-Z0-9._-]*[A-Z0-9])/i
```

Comment:
I think this needs the first part of the regex too, or else we won't be able to parse single character packages like [a](https://pypi.org/project/a/)

Reply:
I think it would fail on something like `a>=3`, for example

Reply:
oooh you're right!

Reply:
good ol' a